pkg/server/routes/snapshots.go

```diff
@@ -0,0 +1,113 @@
+package routes
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/pkg/config"
+	"github.com/loft-sh/vcluster/pkg/snapshot"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	PostSnapshotsURL = "/vcluster/snapshots"
+)
+
+func WithSnapshotsCreate(ctx context.Context, uncachedLocalClient, uncachedVirtualClient client.Client, vConfig *config.VirtualClusterConfig) http.Handler {
+	logger := log.GetInstance()
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method != http.MethodPost {
+			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		// read the snapshot request and options from the HTTP request body
+		body, err := io.ReadAll(req.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		var snapshotRequestData map[string]json.RawMessage
+		err = json.Unmarshal(body, &snapshotRequestData)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		var snapshotRequest snapshot.Request
+		err = json.Unmarshal(snapshotRequestData[snapshot.RequestKey], &snapshotRequest)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		var snapshotOptions snapshot.Options
+		err = json.Unmarshal(snapshotRequestData[snapshot.OptionsKey], &snapshotOptions)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		// create the snapshot request Secret and ConfigMap
+		// - for shared nodes, create resources in the host cluster in the vCluster namespace
+		// - for private nodes, create resources in the virtual cluster in the kube-system namespace
+		var kubeClient client.Client
+		var snapshotRequestNamespace string
+		if vConfig.Config.PrivateNodes.Enabled {
+			kubeClient = uncachedVirtualClient
+			snapshotRequestNamespace = "kube-system"
+		} else {
+			kubeClient = uncachedLocalClient
+			snapshotRequestNamespace = vConfig.HostNamespace
+		}
+
+		// create snapshot request Secret
+		secret, err := snapshot.CreateSnapshotOptionsSecret(snapshotRequestNamespace, &snapshotOptions)
```

Comment:
Not sure if it's out of scope, but I think it would be good to have some kind of rate-limiting mechanism, either here or in the CLI. Basically to reject a request for creating a new snapshot if the last one is not done yet. WDYT?